compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -526,7 +529,7 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
             }),
             ty::FnPtr(_) => {
                 let mut ptr = scalar_unit(Pointer);
-                ptr.valid_range = 1..=*ptr.valid_range.end();
+                ptr.valid_range = AllocationRange { start: 1, end: ptr.valid_range.end };
```

Comment:
This pattern probably deserves a `with_start` method that takes a range and replaces the start.

Reply:
Will implement it and `with_end`